### ISSUE TITLE
Courier cannot deliver Divine Rapier from stash on Turbo

### ISSUE THREAD

**Issue by StartyDotty** [ValveSoftware/Dota2-Gameplay#3077]
### Item name

Divine Rapier

### Description

- I buy Divine Rapier at 23:42 in this turbo game
    - The courier is in base, not the secret shop.
    - I am not standing in the secret shop.
    - The Rapier stays in my stash
- The courier rushes to deliver an item to me, but the Rapier stays in my stash.
- The courier continues to go back and forth between me and my base, delivering nothing.

Considering the fact that in Turbo your normal shop is supposed to act exactly as a Secret Shop, it stands to reason that a courier should be able to deliver a Rapier from your stash in Turbo.

### Example Match ID and Timestamp

6756797027 @23:42

**Comment by commentator24** [comment]
not a bug.
courier can buy rapier parts, combine into rapier, but once its removed from its inventory it can no longer pick it up. e.g. from ground, stash, other players (including the owner). try buy rapier with courier, drop it to other player / ground / your hero to place back in courier, wont work.

removing rapier from its inventory for whatever reason, e.g. to Stash because you died / deliver to stash first etc means it can no longer hold rapier, so it will deliver nothing.
I would suggest to simply buy rapier parts and lock one of the parts, combine later at base. its much safer in ranked games too

**Comment by achoooooo** [comment]
Please read my post in more detail, I do think this is a bug.

> courier can buy rapier parts, combine into rapier

This is exactly what isn't working. My courier is sitting in base when I attempt to buy Rapier, which the normal shop does sell.

> once its removed from its inventory it can no longer pick it up

It doesn't ever entry my inventory, the Rapier never leaves the stash.

**Comment by commentator24** [comment]
when you buy rapier parts in turbo, u buy from the shop in base, which automatically goes to Stash

so because it already combine into rapier in stash, courier cannot pick it up from stash because its not allowed to. this would be different story if you buy relic/demon edge first, put into courier then buy the other rapier part. because this combines rapier in courier inventory not stash. do you get it?

sorry if it sounds rude, but i dont think this is a bug, should close thread

**Comment by achoooooo** [comment]
> courier cannot pick it up from stash because its not allowed to

Why is a courier not allowed to pick up a rapier from stash? That is the bug. There’s no reason why a courier shouldn’t be able to *take it out of the stash*. But I understand that *putting it back in* should be disallowed.

So the issue is that the courier is completely disallowed from working with the stash due to the assumption that in normal dota a Rapier couldn’t easily end up there because it must be purchased in the secret shop. But in Turbo that isn’t the case, rapiers are very likely to end up in your stash because that’s the default place purchased items go to.

**Comment by commentator24** [comment]
well, if you want to make courier deliver rapier you only have 2 options.
1. Select courier first before buying rapier. this will make bought parts go to your courier inventory not stash. Hence it will combine in courier. 
2. buy and lock one of rapier parts so it doesn't combine into rapier before delivering it to you. this is the safest way.

once you get used to it, its not that difficult

**Comment by achoooooo** [comment]
> well, if you want to make courier deliver rapier you only have 2 options.
> …
> once you get used to it, its not that difficult


Yes, I understand that it’s still possible. But that doesn’t mean this isn’t a bug. “Get used to it” isn’t a helpful reply, this forum is meant to track issues in Dota - which this is.

**Comment by lillemessiax** [comment]
Hey, first time playing Dota 2, but basically the same thing.
Purchased two rapiers at like 15 min, but courier went back and forth.
If it's like commentator24 is saying: courier can never hold the completed item, I'm not sure if its a bug or a feature, but still really annoying.
Is the solution to go back to base to pick them up yourself if they get stack in stash?
